grails-app/migrations/views/cycle-count-session.sql

```diff
@@ -47,36 +47,48 @@ SELECT
+                          FROM inventory_level
+                          WHERE inventory_level.internal_location_id IS NULL
+                          GROUP BY inventory_id, product_id) as inventory_level_summary
                          ON product_availability.product_id = inventory_level_summary.product_id AND
                             location.inventory_id = inventory_level_summary.inventory_id
 
-         LEFT OUTER JOIN (select cycle_count_request.facility_id                          as facility_id,
+    -- The following
+         LEFT OUTER JOIN (SELECT cycle_count_request.facility_id                          as facility_id,
                                  cycle_count_request.product_id                           as product_id,
                                  cycle_count_request.id                                   as cycle_count_request_id,
                                  COALESCE(cycle_count.status, cycle_count_request.status) as status
-                          from cycle_count_request
-                                   left outer join cycle_count on cycle_count_request.cycle_count_id = cycle_count.id
-                          where cycle_count_request.id IS NULL
+                          FROM cycle_count_request
+                                   LEFT OUTER JOIN cycle_count on cycle_count_request.cycle_count_id = cycle_count.id
+                          WHERE cycle_count_request.id IS NULL
                              OR (cycle_count_request.status <> 'COMPLETED'
                               AND cycle_count_request.status <> 'CANCELED')) as cycle_count_request_summary
-                         on cycle_count_request_summary.product_id = product_availability.product_id
-                             and cycle_count_request_summary.facility_id = product_availability.location_id
+                         ON cycle_count_request_summary.product_id = product_availability.product_id
+                             AND cycle_count_request_summary.facility_id = product_availability.location_id
 
+         LEFT OUTER JOIN (SELECT inventory_id,
+                                 product_id,
+                                 abc_class,
+                                 max(date_expected)         as date_expected,
+                                 max(days_until_next_count) as days_until_next_count
+                          FROM cycle_count_metadata
+                          GROUP BY inventory_id, product_id) as cycle_count_metadata
+                         on cycle_count_metadata.inventory_id = location.inventory_id and
+                            cycle_count_metadata.product_id = product_availability.product_id
 
 GROUP BY product_availability.location_id, product_availability.product_id, location.inventory_id
-HAVING sum(product_availability.quantity_on_hand) > 0
-ORDER BY NULL
+HAVING sum(product_availability.quantity_on_hand) > 0 and cycle_count_metadata.days_until_next_count <= 0
```

Comment:
These two conditions need to be moved to the candidate query since this will cause the records to disappear from tabs other than All Products.